arelle/plugin/validate/DBA/rules/__init__.py

```diff
@@ -218,3 +218,11 @@ def getFactsGroupedByContextId(modelXbrl: ModelXbrl, *conceptQns: QName) -> dict
             groupedFacts[contextId] = []
         groupedFacts[contextId].append(fact)
     return dict(sorted(groupedFacts.items()))
+
+
+def lookup_namespaced_facts(modelXbrl: ModelXbrl, namespaceURI: str) -> set[ModelFact]:
+    """
+    Returns the set of facts that are tagged with a concept from a particular namespace
+    :Return: a set of facts
+    """
+    return {f for f in modelXbrl.facts if f.concept.qname.namespaceURI == namespaceURI and f.xValid >= VALID and f.xValue is not None}
```

Comment:
I'm not sure if it's needed, but putting the xValid check before the namespace check might prevent an exception from being raised on invalid docs.